### PR TITLE
Fix attendee block query

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR:
- Adds support for `event` param for attendee query
- Fixes the defaults for the block attributes

### CODE REVIEW

**Comment by manzoorwanijk** [comment]
@tn3rb I have added support for Datetime query param, please have a look

File: assets/src/domain/blocks/event-attendees/attributes.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ const attributes: Block<EventAttendeesAttributes>['attributes'] = {
 	},
 	orderBy: {
 		type: 'string',
-		default: 'lastThenFirstName',
+		default: 'LAST_THEN_FIRST_NAME',
```
Comment (nicolad) [inline assets/src/domain/blocks/event-attendees/attributes.ts:32]: would be great to have enums for these

Comment (manzoorwanijk) [reply assets/src/domain/blocks/event-attendees/attributes.ts:32]: String literals are even better, they don't require imports and still guard the types.

Comment (nicolad) [reply assets/src/domain/blocks/event-attendees/attributes.ts:32]: didn't noticed that `Block<EventAttendeesAttributes>['attributes']`, agree this is more convenient

File: assets/src/data/services/apollo/queries/attendees/types.ts [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
 	orderby?: EntityQueryOrderBy<AttendeesOrderByFields>;
 	regStatus?: RegistrationStatus;
-	regTicket?: string;
-	regTicketId?: number;
-	regTicketIdIn?: Array<number>;
-	regTicketIn?: Array<string>;
+	regTicket?: EntityId;
+	regTicketId?: EntityDbId;
+	regTicketIdIn?: Array<EntityDbId>;
+	regTicketIn?: Array<EntityId>;
```
Comment (tn3rb) [inline assets/src/data/services/apollo/queries/attendees/types.ts:15]: much better 👍

File: assets/src/domain/blocks/services/apollo/queries/useAttendees.ts [hunk 3]
```diff
@@ -27,6 +27,9 @@ const useAttendees: UseAttendees = (where, limit) => {
 			where,
 		},
 		fetchPolicy: 'cache-first',
+		// do not fetch until an event has been selected
+		// otherwise, we will end up fetching unrelated attendees
+		skip: !where.event,
```
Comment (tn3rb) [inline assets/src/domain/blocks/services/apollo/queries/useAttendees.ts:32]: noice 👌

File: core/domain/services/graphql/connection_resolvers/AttendeeConnectionResolver.php [hunk 4]
```diff
@@ -99,14 +100,43 @@ public function get_query_args()
...
             if (! empty($input_fields['Registration.Ticket.TKT_ID']) && is_array($input_fields['Registration.Ticket.TKT_ID'])) {
-                $input_fields['Registration.Ticket.TKT_ID'] = ['in', $input_fields['Registration.Ticket.TKT_ID']];
+                $input_fields['Registration.Ticket.TKT_ID'] = ['IN', $input_fields['Registration.Ticket.TKT_ID']];
             }
-            // Use the proper operator.
-            if (! empty($input_fields['Registration.Event.EVT_ID']) && is_array($input_fields['Registration.Event.EVT_ID'])) {
-                $input_fields['Registration.Event.EVT_ID'] = ['in', $input_fields['Registration.Event.EVT_ID']];
+            // If Ticket param is passed, it will have preference over Datetime param
+            // So, use Datetime param only if a Ticket param is not passed
+            if (! empty($input_fields['Datetime.DTT_ID']) && empty($input_fields['Registration.Ticket.TKT_ID'])) {
```
Comment (tn3rb) [inline core/domain/services/graphql/connection_resolvers/AttendeeConnectionResolver.php:116]: 👍

File: core/domain/services/graphql/connection_resolvers/AttendeeConnectionResolver.php [hunk 5]
```diff
@@ -99,14 +100,43 @@ public function get_query_args()
...
+                try {
+                    // Get related ticket IDs for the given dates
+                    $ticketIds = EEM_Ticket::instance()->get_col([
+                        [
+                            'Datetime.DTT_ID' => ['IN', $datetimeIds],
+                            'TKT_deleted'     => ['IN', [true, false]],
+                        ],
+                        'default_where_conditions' => 'minimum',
+                    ]);
+                } catch (\Throwable $th) {
```
Comment (tn3rb) [inline core/domain/services/graphql/connection_resolvers/AttendeeConnectionResolver.php:130]: plz import all classes, ie: add use statement at top of class for this, then simplify FQCN here

Comment (manzoorwanijk) [reply core/domain/services/graphql/connection_resolvers/AttendeeConnectionResolver.php]: Done

**Pull request merged**